actionpack/lib/action_dispatch/http/content_security_policy.rb

```diff
@@ -198,6 +200,20 @@ def require_sri_for(*types)
       end
     end
 
+    def trusted_types(*values)
+      if values.first
+        if values.first.eql? :none
+          values << "'#{values.pop}'"
+        elsif values.last.eql? :allow_duplicates
+          values << "'#{values.pop.to_s.tr('_', '-')}'"
+        end
```

Comment:
I assume that you're trying to prevent invalid definitions with this block? I don't think we need to do that - we aren't capturing all of the possible invalid scenarios and the extra dynamic string objects per request isn't worth it, so just use the standard method.

Reply:
Done. About deprecated csp headers like report-uri and plugin-types its not interesting add a deprecation msg or even remove them?

Reply:
> Done. About deprecated csp headers like report-uri and plugin-types its not interesting add a deprecation msg or even remove them?

Possibly, but we'd need to check browser usage/share - this it meant to be a lightweight DSL and not a validator.

Reply:
For sure, In other day I'll try to take a look into this. Thanks for guidance.